src/html/scan.js

```diff
@@ -552,7 +552,6 @@ function updateOptions(data) {
   if (data['wifi-ssid']) _('homenet').textContent = data['wifi-ssid'];
   else _('connect').style.display = 'none';
   if (data['customised']) _('reset-options').style.display = 'block';
-  _('customised').value = true; // overwrite the value from the module, if we save then it's customised
```

Comment:
I had to add this because we default to `customised` false. Then when it loads the json it will set the hidden field value to false. Then when we "save & reboot", it just uploads the new json file with customised set to false.

Reply:
Ah fudge me, you're absolutely right. Would it make more sense to maybe remove the hidden field entirely and just slap customized = true into the POST javascript rather than trying to keep forcing the form value to a fixed value?
```diff
diff --git a/src/html/index.html b/src/html/index.html
index 6a20511b..93ef6641 100644
--- a/src/html/index.html
+++ b/src/html/index.html
@@ -41,7 +41,6 @@
                                                <label for="phrase">Binding Phrase</label>
                                        </div>
                                        <form id='upload_options' method='POST' action="/options">
-                                               <input type="hidden" id="customised" name="customised" value="true"/>
                                                <input type="hidden" id="wifi-ssid" name="wifi-ssid"/>
                                                <input type="hidden" id='wifi-password' name='wifi-password'/>
                                                <div class="mui-textfield">
diff --git a/src/html/scan.js b/src/html/scan.js
index 175e3d38..ef57ade2 100644
--- a/src/html/scan.js
+++ b/src/html/scan.js
@@ -470,6 +470,8 @@ function submitOptions(e) {
   const formObject = Object.fromEntries(new FormData(formElem));
   // Add in all the unchecked checkboxes which will be absent from a FormData object
   formElem.querySelectorAll('input[type=checkbox]:not(:checked)').forEach((k) => formObject[k.name] = false);
+  // Force customised to true as this is now customising it
+  formObject['customised'] = true;

   // Serialize and send the formObject
   xhr.send(JSON.stringify(formObject, function(k, v) {
```

Reply:
Agreed, the other option is to change line 223 (saveOptions) to
```
if (customised)
{
    doc["customised"] = customised;
}
```
That way we only have the flag if it's actually true!

Reply:
Thanks for your help with this, PK. Yeah I guess it could be not set on the firmware side too. I pushed that change I proposed. haha this was such a small cosmetic thing and we've made a week out of it!

Reply:
Something about cans and worms 😉